### ISSUE TITLE
"Abort trap: 6" on run.sh

### ISSUE THREAD

**Issue by apierre-suse** [backslash-f/ShockEmu#9]
Hey,

On MacBook Air M1 2020 running macOS Big Sur 11.5:
- System Integrity Protection status: disabled
- build.sh has been changed: python3 shockemu.py $1
- Running the build gives a warning but is creating new iohid_wrap.dylib, iohid_wrap.m and mapKeys.h:

```
iohid_wrap.m:64:35: warning: cast to smaller integer type 'unsigned int' from 'IOHIDDeviceRef _Nonnull' (aka 'struct __IOHIDDevice *') [-Wpointer-to-int-cast]
        printf("IOHIDDeviceOpen %08x\n", (unsigned int) device);
                                         ^~~~~~~~~~~~~~~~~~~~~
1 warning generated.
```

However, when I execute the run script, I get:

`./run.sh: line 3:  3172 Abort trap: 6           DYLD_FORCE_FLAT_NAMESPACE=1 DYLD_INSERT_LIBRARIES=iohid_wrap.dylib /Applications/RemotePlay.app/Contents/MacOS/RemotePlay > /dev/null 2>&1`

For information, I can manually start RemotePlay using this command: `/Applications/RemotePlay.app/Contents/MacOS/RemotePlay`

PS: is this soft only working with RemotePlay? How can I make it to work with any other app using the PS4 DualShock?

Thanks for your help!

**Comment by backslash-f** [comment]
I'm getting the exact same error on macOS 12 (not a M1 machine). It's not M1 related. I don't know how to fix it yet.

> is this soft only working with RemotePlay?

Yes.

> How can I make it to work with any other app using the PS4 DualShock?

It may be possible. I've never tried myself. Maybe start by adding the app you want to control in `run.sh`, instead of `/Applications/RemotePlay.app/Contents/MacOS/RemotePlay`. But I'm not sure that would work.

**Comment by backslash-f** [comment]
(BTW that warning was just fixed in master.)

**Comment by backslash-f** [comment]
Could you try again with latest master? I just pushed some changes. 👆🏻

**Comment by apierre-suse** [comment]
I just tried again. The first time SIP was enabled and I could build and run but my keyboard wasn't interacting as a DualShock on the RemotePlay app. So I disabled SIP again and now I'm facing the same error:

`./run.sh: line 3:  2915 Abort trap: 6           DYLD_FORCE_FLAT_NAMESPACE=1 DYLD_INSERT_LIBRARIES=iohid_wrap.dylib /Applications/RemotePlay.app/Contents/MacOS/RemotePlay > /dev/null 2>&1`

**Comment by backslash-f** [comment]
Thanks for checking. I'll try again later on when I'm at home.

**Comment by backslash-f** [comment]
Yeah, no dice. Still crashing on me too... Let's keep this one open so others can jump in. 🔨

**Comment by backslash-f** [comment]
Depends on #1
